Client/Frontend/Settings/AppSettingsTableViewController.swift

```diff
@@ -78,6 +78,27 @@ class AppSettingsTableViewController: SettingsTableViewController {
             accountSectionTitle = NSAttributedString(string: Strings.FxAFirefoxAccount)
         }
 
+        let engineSettingChanged: ((String)->((Bool)->Void)) = { engineName in
+            let prefName = "sync.engine." + engineName + ".enabledStateChanged"
+            return { enabled in
+                if let _ = prefs.boolForKey(prefName) { // Switch it back to not-changed
+                    prefs.removeObjectForKey(prefName)
+                } else {
+                    prefs.setBool(true, forKey: prefName)
+                }
+                // We trigger a quick sync of the Clients collection in order to spin the Sync state machine
+                // which will update meta/global.
```

Comment:
I think an ideal choice here is:

- If you just flipped from false to true, sync _the selected engine_ immediately. It might even be ready by the time we leave this UI!
- If you flipped from true to false, sync _the deselected engine_ when the user leaves the Settings screen. That way if they accidentally uncheck History, we don't immediately delete ten megs of data from the server.